### ISSUE TITLE
windows_virtual_machine validation assumes computer_name is not set

### ISSUE THREAD

**Issue by adamday2** [terraform-providers/terraform-provider-azurerm#6494]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
Terraform v0.12.21
AzureRM 2.5.0

### Affected Resource(s)

* `azurerm_windows_virtual_machine`

### Terraform Configuration Files

```hcl
resource "azurerm_windows_virtual_machine" "this" {
   name          = "longnameover15chars"
   computer_name = "shortname"
...
}
```

### Expected Behavior

No errors, the `name` in the portal is a valid name for the resource in the portal (there is no 15 character limit) and the `computer_name` is a valid hostname.

### Actual Behavior

```
Error: "name" can be at most 15 characters, got 19

  on .terraform/modules/server/main.tf line 112, in resource "azurerm_windows_virtual_machine" "this":
 112: resource "azurerm_windows_virtual_machine" "this" {
```

It appears the the validation function is unaware of whether the `name` is being used as the `computer_name` and is applying the same rules regardless. In the case that the `computer_name` is provided, the rules should not be checking for valid hostnames, but instead just validating that it is a valid azure resource name.

https://github.com/terraform-providers/terraform-provider-azurerm/blob/18e516bd25aed3c99477208aa2486272d7214024/azurerm/internal/services/compute/windows_virtual_machine_resource.go#L55

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

Azure Government

### References

* #0000

**Comment by njuCZ** [comment]
@adamday2 thanks for pointing this problem, it has been fix by #6639, which is released by version 2.8.0. Could you please upgrade your azurerm version ?

**Comment by tombuildsstuff** [comment]
hi @adamday2 

As @njuCZ has mentioned this has been fixed in v2.8.0 - you can upgrade to this by updating the version in the Provider block (as shown below) and then running `terraform init -upgrade`:

```
provider "azurerm" {
  version = "= 2.8.0"
}
```

Since this should be fixed by updating the version of the Azure Provider being used I'm going to close this issue for the moment - but please let us know if that doesn't work for you and we'll take another look.

Thanks!

**Comment by adamday2** [comment]
Appears to be fixed now. Thank you!
